rfc-0001/README.md

```diff
@@ -0,0 +1,331 @@
+directory is the absolute source of truth.
+
+## Evolution begins here
+
+The above schema works if we never need to migrate a (production) system with data in place.
+However, this expectation is neither realistic nor practical.
+
+When we are ready to capture a revision of the schema (including the very first one), we should start
+revision identification so we can easily refer to them in the future.
+
+Let's use UUID v7 (time-ordered) identifiers. They fit UUID format, they are sorted and convey the (non-strict) notion of time.
+
+Now, in order for revisions to be properly ordered, without relying on time, let's propose that every revision can refer to
+zero or more parent revisions. There could be only one revision with zero parents (the very first one) and we should always
+make revisions converge to a linear ending by having one "interim" final revision so we don't have much divergence. We'll cover that later.
+
+For better comprehension, let's visualize such revisions as directories "revisions" directory, named after their ID:
+
+```
+0194f2ae-5a6f-762a-8ec4-d06d609f14fc/
+```
+
+Since we need to store parent references, let's say every revision will have some
+kind of metadata files:
+
+```
+0194f2ae-5a6f-762a-8ec4-d06d609f14fc/
+  metadata.yaml
+```
+
+### Metadata
+
+**metadata.yaml**
+```yaml
+# To make sure we know what format is this
+"$schema": "https://schema.omnigr.es/revision/v1.json"
+# List of parent IDs this revisions joins
+parents: []
+```
+
+
+> [!IMPORTANT]  
+> Metadata file should not contain any defaults to avoid misinterpretation
+> when the defaults will inadvertently change.
+
+### Sources
+
+Now, since for every revision we should be capturing all the 
+source code involved (to avoid any ambiguity about it), let's capture them all 
+in a single file (to avoid polluting project repository with multitudes of files
+on every revision capture)
+
+```
+0194f2ae-5a6f-762a-8ec4-d06d609f14fc/
+  metadata.yaml
+  sources.dat
+```
+
+**sources.dat** is a provisional name as there are a few formats worth discussing:
+
+| Format           | Pros                                   | Cons                                                                                  |
+|------------------|----------------------------------------|---------------------------------------------------------------------------------------|
+| Tar              | Easy to open using command-line tools  | Not directly understood by Postgres unless copied through a program; uncompressed     |
+| Tar.(gz/bz2/...) | Same as tar + Smaller size             | Same as tar                                                                           |
+| Zip              | Same as tar + Smaller size             | Same as tar                                                                           |
+| SQL              | Can be loaded by sending into Postgres | Can modify arbitrary objects, not directly usable in command-line for file extraction |
```

Comment:
The `SQL` format also seems to be more interesting when using the VCS diff between commits.

Reply:
true, I was mistakenly thinking about diffing between versions (but that would be between files in different revision directories). The point is still valid though, you could generalize `Can be loaded by sending into Postgres` by saying `Easier to inspect`.